python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_asset_check_selection.py

```diff
@@ -0,0 +1,126 @@
+import json
+from pathlib import Path
+
+from dagster import AssetCheckKey, AssetKey
+from dagster._core.definitions.asset_graph import AssetGraph
+from dagster_dbt import (
+    DagsterDbtTranslator,
+    DagsterDbtTranslatorSettings,
+    build_dbt_asset_selection,
+)
+from dagster_dbt.asset_decorator import dbt_assets
+
+dagster_dbt_translator_with_checks = DagsterDbtTranslator(
+    settings=DagsterDbtTranslatorSettings(enable_asset_checks=True)
+)
+
+manifest_path = (
+    Path(__file__)
+    .joinpath("..", "..", "dbt_projects", "test_dagster_asset_checks", "manifest.json")
+    .resolve()
+)
+
+with open(manifest_path, "r") as f:
+    manifest = json.load(f)
+
+
+@dbt_assets(manifest=manifest, dagster_dbt_translator=dagster_dbt_translator_with_checks)
+def my_dbt_assets():
+    ...
+
+
+asset_graph = AssetGraph.from_assets([my_dbt_assets])
+
+ALL_ASSET_KEYS = {
+    AssetKey([key])
+    for key in [
+        "customers",
+        "orders",
+        "raw_customers",
+        "raw_orders",
+        "raw_payments",
+        "stg_customers",
+        "stg_orders",
+        "stg_payments",
+    ]
+}
+
+
+ALL_CHECK_KEYS = {
+    AssetCheckKey(asset_key=AssetKey([asset_name]), name=check_name)
+    for asset_name, check_name in [
+        ("customers", "not_null_customers_customer_id"),
+        ("customers", "unique_customers_customer_id"),
+        (
+            "orders",
+            "accepted_values_orders_status__placed__shipped__completed__return_pending__returned",
+        ),
+        ("orders", "not_null_orders_amount"),
+        ("orders", "not_null_orders_bank_transfer_amount"),
+        ("orders", "not_null_orders_coupon_amount"),
+        ("orders", "not_null_orders_credit_card_amount"),
+        ("orders", "not_null_orders_customer_id"),
+        ("orders", "not_null_orders_gift_card_amount"),
+        ("orders", "not_null_orders_order_id"),
+        ("orders", "relationships_orders_customer_id__customer_id__ref_customers_"),
+        ("orders", "unique_orders_order_id"),
+        ("stg_customers", "not_null_stg_customers_customer_id"),
+        ("stg_customers", "unique_stg_customers_customer_id"),
+        (
+            "stg_orders",
+            "accepted_values_stg_orders_status__placed__shipped__completed__return_pending__returned",
+        ),
+        ("stg_orders", "not_null_stg_orders_order_id"),
+        ("stg_orders", "unique_stg_orders_order_id"),
+        (
+            "stg_payments",
+            "accepted_values_stg_payments_payment_method__credit_card__coupon__bank_transfer__gift_card",
+        ),
+        ("stg_payments", "not_null_stg_payments_payment_id"),
+        ("stg_payments", "unique_stg_payments_payment_id"),
+    ]
+}
+
+
+def test_all():
```

Comment:
Any reason why we're not parametrizing this? All the tests essentially have the same assertion.

Reply:
IMO repeating a couple assert lines is easier to read and adjust than parameterizing test input, output, and name. Parameterizing makes sense for massive suites like AMP but I like keeping it simple when we can